src/posts/2020-07-27-unsucking-your-project-onboarding.md

```diff
@@ -0,0 +1,789 @@
+      4ED778F539E3634C779C87C6D7062848A1AB005C \
+      B9E2F5981AA6E0CD28160D9FF13993A75599653C"
+
+SERVERS="ha.pool.sks-keyservers.net
+         p80.pool.sks-keyservers.net:80 \
+         ipv4.pool.sks-keyservers.net \
+         keyserver.ubuntu.com
+         keyserver.ubuntu.com:80 \
+         pgp.mit.edu
+         pgp.mit.edu:80"
+
+for key in $KEYS; do
+  if ! gpg --list-keys $key 2>&1 >/dev/null; then
+    for server in $SERVERS; do
+      gpg --no-tty --keyserver "hkp://$server" --display-charset utf-8 --recv-keys "$key"  && break
+    done
+  fi
+done
+```
+
+#### Environment setup
+
+Having the languages available, we then move on to setting up the environment
+through environment variables. This usually happens through `dotenv` (less
+common) or (more common nowadays) `direnv`.
+
+Lately all projects I've been on have been built with `direnv` in mind, so let's
+focus on that:
+
+```shell
+pp_info "setup" "Setting up the env..."
+
+if [ ! -f .envrc ]; then
+  cp .envrc.sample .envrc
+  printf "env file created, you might want to open .envrc and set the correct values. We recommend using direnv to manage it.\n\n"
+else
+  printf "envrc file already exists, skipping...\n\n"
+fi
+
+. "./.envrc"
+```
+
+We always add a `.envrc.sample` to the project to document the environment
+variables we need. The condition makes sure no `.envrc` is present when doing
+the copy as to not override it. **We always try to make sure `bin/setup` is
+idempotent**, allowing others to run it repeatedly without making any damages.
+
+There have also been some variations of this snippet that check for the
+installation of `direnv`, similarly to what we do with `asdf`, but it has since
+been dropped.
+
+As a sidenote, if you are using `dotenv`, replace `.envrc` with `.env` and it
+should work!
+
+#### Installing dependencies
+
+In a typical Elixir + React project, this breaks down into three steps: making
+sure `yarn` is available, installing Node dependencies, installing Elixir
+dependencies:
+
+```shell
+pp_info "setup" "Ensuring yarn is installed"
+if not_installed "yarn"; then
+  curl -o- -L https://yarnpkg.com/install.sh | bash
+else
+  echo "All done!"
+fi
+
+pp_info "setup" "Installing node requirements"
+yarn install
+
+pp_info "setup" "Installing elixir dependencies..."
+MIX_ENV=$env mix local.hex --force
+MIX_ENV=$env mix local.rebar --force
+MIX_ENV=$env mix deps.get
+mix archive.install hex phx_new 1.4.8 --force
+```
+
+Digging through different projects, I also found these several variations for
+this step across old projects:
+
+```shell
+# Python project
+pp_info "setup" "Attempting to upgrade pip"
+pip install --upgrade pip
+
+pp_info "setup" "Installing python requirements"
+pip install -r requirements.txt
+
+# Gatsby project (after ensuring the yarn install)
+pp_info "setup" "Ensuring gatsby is installed..."
+if not_installed "gatsby"; then
+  npm install -g gatsby-cli
+else
+  echo "Already installed, skipping."
+fi
```

Comment:
Same reasoning as for Yarn above (you can simply install it and it will be skipped if already installed). However this is only needed for generating new projects, so the same reasoning as install `phx_new` applies here.

```suggestion
npm install -g gatsby-cli
```

Reply:
I don't understand you're point, does this mean you agree it is unnecessary, or do you believe it is necessary?